### PR TITLE
DTSPO-1597 - Sendgrid subusers - 3

### PR DESCRIPTION
**Pull request by chrisorisawayi**
### JIRA link (if applicable) ###
https://tools.hmcts.net/jira/browse/DTSPO-1597

### Change description ###
Name prefix update to subuser password key vault secret
Initial readme update. API key retrieval section to be updated using PR example from DTSPO-1452  once completed.

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[x ] No
```

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1 +1,70 @@
-# sendgrid
\ No newline at end of file
+# Azure SendGrid
+
+This repository provisions SendGrid accounts through the [Azure marketplace](https://azuremarketplace.microsoft.com/en-us/marketplace/apps/sendgrid.sendgrid).
```
Comment (timja) [inline README.md:3]: might be a bit detailed but this repo isn't creating the original accounts
```suggestion
This repository provisions SendGrid sub user accounts in our manually created SendGrid accounts, 'sendgridnonprod' and 'sendgridprod', provisioned through the [Azure marketplace](https://azuremarketplace.microsoft.com/en-us/marketplace/apps/sendgrid.sendgrid).
```

File: README.md [hunk 2]
```diff
@@ -1 +1,70 @@
...
+
+## Setting up a new account
+
+Each team can create a non-production and a production account.  Accounts are created as subusers under the respective non-production and production primary accounts.
```
Comment (timja) [inline README.md:7]: not needed imo
```suggestion
Each team can create a non-production and a production account.  Accounts are created as subusers under the respective non-production and production accounts.
```

File: README.md [hunk 3]
```diff
@@ -1 +1,70 @@
...
+
+You can create new accounts by updating the configuration in 
+[nonprod.tfvars](config/nonprod.tfvars) and [prod.tfvars](config/prod.tfvars)
+
+Accounts are created with an 'hmcts-' prefix appended to avoid name clashes.
```
Comment (timja) [inline README.md:12]:
```suggestion
Accounts are created with an 'hmcts-' prepended to avoid name clashes.
```

File: README.md [hunk 4]
```diff
@@ -1 +1,70 @@
...
+
+Then tick 'I've added these records.' and click 'Verify'.
+
+If successful then DMARC is all setup.
+
+## API key
+
+SendGrid requires an API key for sending emails through it.
+
+An API key is automatically created same time as the account and stored in the SendGrid Key Vault.  Key has restricted access with 'Mail Send' permission allowed.
```
Comment (timja) [inline README.md:65]:
```suggestion
An API key is automatically created for you when the account is created and stored in the SendGrid Key Vault. The key has restricted access with 'Mail Send' permission allowed.

The API key name format is `hmcts-<account-name>-api-key`
```

File: README.md [hunk 5]
```diff
@@ -1 +1,70 @@
...
+
+
+### Getting the key from the application
+
+The API key should be read from Key Vault via terraform.
```
Comment (timja) [inline README.md:70]: should link the example from the previous repo for now and you can update it with divorce when you do it?

**Review by timja — COMMENTED** [review]
looks good, few comments

File: README.md (hunk 5 above)
Comment (chrisorisawayi) [reply README.md:72]: sure, now added.

File: terraform/10-sendgrid.tf [hunk 6]
```diff
@@ -28,7 +28,7 @@ resource "azurerm_key_vault_secret" "subuser" {
   for_each = { for user in var.accounts : user.name => user }
 
 
-  name         = "${each.value.name}-password"
+  name         = "hmcst-${each.value.name}-password"
```
Comment (timja) [inline terraform/10-sendgrid.tf:31]:
```suggestion
  name         = "hmcts-${each.value.name}-password"
```

**Review by timja — APPROVED** [review]

**Pull request merged**